frost-core/src/frost/round1.rs

```diff
@@ -314,14 +328,14 @@ where
 /// Generates the signing nonces and commitments to be used in the signing
 /// operation.
 ///
-/// [`commit`]: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-04.html#name-round-one-commitment
+/// [`commit`]: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-05.html#section-5.1
 pub fn commit<C, R>(
-    participant_index: u16,
+    key_package: &KeyPackage<C>,
```

Comment:
I'm tempted to recommend making the items that we use in KeyPackage explicit: `participant_index` (which should be identifier soon), and the `signing_nonces` which can be extracted by methods (and probably should be in case we make those fields private

Reply:
This is mostly because `KeyPackage` is a software conceit and `commit()` is from the spec so

Reply:
> You mean, using `participant_index` and `secret` as parameters instead of `KeyPackage`? I was thinking about encapsulating things to avoid having the user think about how to get the index and the secret. But I don't mind either way. Arguably we could get rind of index since it's not used here, but that would require some refactoring since it's used later.

Yeah basically, it's not great but it's more explicit and closer to the spec, and does not pass in extra stuff in the `KeyPackage` the function doesn't need


> Also, I just realized that I should change the method to return a single `SigningNonces`/`SigningCommitments` instead of a vector. What do you think?

Oh yeah good catch, agreed